### PR TITLE
fix: rename `aarch64` to `arm64` on Linux.

### LINKED ISSUE #1: 404 error on Linux arm64.
**Describe the bug**

This package detects the arch as `aarch64` however releases of flyctl are for `arm64`.

**Steps to reproduce**

I'm using an ubuntu-based devcontainer on my M1 mac:

```
$ asdf install flyctl latest
mkdir: cannot create directory ‘/home/vscode/.asdf/downloads/flyctl/0.0.464’: File exists
* Downloading flyctl release 0.0.464...
curl: (22) The requested URL returned error: 404
asdf-flyctl: Could not download https://github.com/superfly/flyctl/releases/download/v0.0.464/flyctl_0.0.464_Linux_aarch64.tar.gz
```

**Expected behavior**

flyctl should be installed.

### PR DESCRIPTION
**Pull request by jimsynz**
Closes #1.

### CODE REVIEW

File: lib/utils.bash [hunk 1]
```diff
@@ -80,7 +80,17 @@ get_platform() {
...
+  platform=$(uname -s)
+
+  local arch
+  arch=$(uname -m)
+
+  if [[ "$platform" = "Linux" && "$arch" == "aarch64" ]]; then
+    echo "arm64"
+  else
+    echo "$arch"
+  fi
```
Comment (chessmango) [inline lib/utils.bash:93]: Given this applies to Mac too (though [Wikipedia's uname examples](https://en.wikipedia.org/wiki/Uname) suggest `arm64`) we can probably afford to be broad here and avoid an extra platform check:

```suggestion
  local arch
  arch=$(uname -m)

  if [[ "$arch" == "aarch64" ]]; then
    echo "arm64"
  else
    echo "$arch"
  fi
```

**Review by chessmango — CHANGES_REQUESTED** [review]
Thanks for the PR! Glad to see this is in use somewhere as well :)
Just suggesting being less specific for ease, and then in case something other than Linux reports `aarch64` for whatever reason.

**Comment by jimsynz** [comment]
Thanks man. Yeah, I use ASDF in all my devcontainers because we can manage the environment from a single tool versions file. Had a client that deploys via fly - easy enough to add to the container this way.

**Comment by chessmango** [comment]
> Thanks man. Yeah, I use ASDF in all my devcontainers because we can manage the environment from a single tool versions file. Had a client that deploys via fly - easy enough to add to the container this way.

Very cool. I'm trying to find ways to have my org do the same, but plugin quality varies pretty wildly. Mine probably aren't the greatest either ;) though I try to keep things consistent at least

Thanks again for the PR!

**Pull request merged**